.github/workflows/linux.yml

```diff
@@ -59,22 +62,27 @@ jobs:
           QT_PKGS: ${{ matrix.qt-version-major == '6' && 'qt6-base-private-dev libgl-dev' || 'qtbase5-private-dev' }}
         run: |
           sudo apt update
-          sudo apt install -y build-essential qtchooser ${{ env.QT_PKGS }}
+          sudo apt install -y build-essential qtchooser ${{ env.QT_PKGS }} cmake ninja
+
+      - name: Install Qt 6 system dependencies
+        if: startsWith(matrix.qt-version, '6.')
+        run: |
+          sudo apt update
+          sudo apt install -y libxkbcommon-dev
 
       - name: Setup GCC problem matcher
         if: matrix.build-type == 'debug'
         uses: ammaraskar/gcc-problem-matcher@0.2.0
 
       - name: Build (${{ matrix.build-type }})
         shell: bash
+        env:
+          CMAKE_PREFIX_PATH: ${{ matrix.qt-version-major == '6' && env.Qt6_DIR || env.Qt5_DIR }}
         run: |
-          mkdir build
-          cd build
-          # Workaround for qtchooser not supporting Qt 6: https://bugs.launchpad.net/ubuntu/+source/qtchooser/+bug/1964763
-          echo -e '/usr/lib/qt6/bin\n/usr/lib/x86_64-linux-gnu' | sudo tee /usr/lib/x86_64-linux-gnu/qtchooser/qt6.conf
-          export QT_SELECT=qt${{ matrix.qt-version-major }}
-          qmake -config ${{ matrix.build-type }} ../src/Notes.pro
-          make -j$(nproc)
+          mkdir build && cd build
```

Comment:
Nit:

```suggestion
          mkdir build
          cd build
```

Reply:
What's the motivation behind this?

Reply:
The `&&` check is redundant. If `mkdir` fails, the whole step will also fail immediately.